plugin.rb

```diff
@@ -72,6 +72,8 @@
                 reviewable.perform(guardian.user, :confirm_spam)
               end
             end
+        elsif opts[:delete_posts]
+          ReviewableAkismetPost.where(target_created_by: user).destroy_all
```

Comment:
Bare with me as I have a lack of context for how this plugin is implemented, I'm wondering why can't add an ActiveRecord association to the `Post` model such that `ReviewableAkismetPost` is destroyed when the relevent posts are destroyed?

Reply:
@tgxworld because the `delete_posts` flag will only [soft deleting the posts](https://github.com/discourse/discourse/blob/e040de0c2cc7c853d95572350b00e56f0d748731/app/services/user_destroyer.rb#L129). We can't destroy the `ReviewableAkismetPost` record when the post is soft deleted. Because sometimes when we mark the posts as not spam, we recover the soft deleted posts.